openassessment/xblock/ui_mixins/mfe/mixin.py

```diff
@@ -62,61 +63,49 @@ def get_block_info(self, data, suffix=""):  # pylint: disable=unused-argument
-            else:
-                raise JsonHandlerError(400, f"Cannot jump to step: {jump_step}")
+        active_step = suffix
 
-        # Determine which mode we are viewing in, since data comes from different sources
-        # View submission, submitted or draft
-        if workflow_step == "submission" or jump_step == "submission":
-            serializer_context.update({"view": "submission"})
-        # View the current assessment step
-        else:
-            serializer_context.update({"view": "assessment"})
+        # Validate that any active step is a valid step
+        if suffix and suffix not in handler_suffixes.STEP_SUFFIXES:
+            raise OraApiException(400, error_codes.INCORRECT_PARAMETERS, f"Invalid step name: {active_step}")
 
-        page_context = PageDataSerializer(self, context=serializer_context)
-        return page_context.data
+        serializer_context = {"active_step": None, "workflow_step": workflow_step}
 
-    def _can_jump_to_step(self, workflow_step, jump_step):
-        """ A helper to determine if we can jump to a step or not """
+        # For the general case, just return refreshed page data, without a response
+        if not active_step:
+            return PageDataSerializer(self, context=serializer_context).data
 
-        step_name_mappings = {
-            "submission": "submission",
-            "peer": "peer",
-            "grades": "done"
-        }
+        # Check to see if user can access this step
+        if not self._user_has_reached_step(workflow_step, active_step):
+            raise OraApiException(400, error_codes.INACCESSIBLE_STEP, f"Inaccessible step: {active_step}")
+
+        # If they have access to this step, return the associated data
+        serializer_context.update({"active_step": active_step})
+        return PageDataSerializer(self, context=serializer_context).data
+
+    def _user_has_reached_step(self, workflow_step, step):
+        """ A helper to determine if we can jump to a step or not """
 
-        workflow_step_to_jump_to = step_name_mappings[jump_step]
+        # Map MFE step name to workflow step name
+        requested_workflow_step = STEP_NAME_MAPPINGS[step]
 
-        # Can always "jump" to submission
-        if workflow_step_to_jump_to == "submission":
+        # Submission is start state, have always reached this
+        if requested_workflow_step == "submission":
             return True
 
-        # Can always "jump" to the step you're on
-        if jump_step == workflow_step:
+        # Have reached your current workflow step
+        if requested_workflow_step == workflow_step:
             return True
 
-        # Can jump to a step you've completed
-        step_status = self.workflow_data.status_details.get(workflow_step_to_jump_to, {})
+        # Have reached any step you have completed
+        step_status = self.workflow_data.status_details.get(requested_workflow_step, {})
```

Comment:
workflow api method that takes a step name and returns whether or not it has been completed?